### PR TITLE
week-5 assignment

### PR DESCRIPTION
**Pull request by shubhamkar-suryawanshi**
@veer-auro Please Review

### CODE REVIEW

**Comment by veer-auro** [comment]
@shubhamkar-suryawanshi Wheres the question 4 of assignment

File: Week-5/assignment1.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+init();
+
+// Generator
+
+function* simpleGenerator() {
```
Comment (veer-auro) [inline Week-5/assignment1.js:46]: you need to call doTask1, doTask2, doTask3 using generator

File: Week-5/assignment1.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  return new Promise((resolve, reject) => {
+    console.log(` ${makePayment}`);
+    if (makePayment) resolve("Order Confirmed");
+    else reject("kindly pay the bill");
+  });
+}
+
+function doTask3(showOrderDetails) {
+  return new Promise((resolve, reject) => {
+    console.log(`${showOrderDetails}`);
```
Comment (veer-auro) [inline Week-5/assignment1.js:21]: Maybe randomly resolve reject here just to demonstrate all 3 states of promise

**Review by veer-auro — CHANGES_REQUESTED** [review]

**Comment by veer-auro** [comment]
@shubhamkar-suryawanshi Left some comments. Please resolve

**Comment by shubhamkar-suryawanshi** [comment]
ok sir

**Pull request merged**